sql/core/src/test/scala/org/apache/spark/sql/DataFrameWindowFunctionsSuite.scala

```diff
@@ -700,6 +700,44 @@ class DataFrameWindowFunctionsSuite extends QueryTest
         Row("b", 3, null, null, null)))
   }
 
+  test("lead/lag with ignoreNulls") {
+    val nullStr: String = null
+    val df = Seq(
+      ("a", 0, nullStr),
+      ("a", 1, "x"),
+      ("a", 2, "y"),
+      ("a", 3, "z"),
+      ("a", 4, nullStr),
+      ("b", 1, nullStr),
+      ("b", 2, nullStr)).
+      toDF("key", "order", "value")
+    val window = Window.orderBy($"order")
+    checkAnswer(
+      df.select(
+        $"key",
+        $"value",
+        $"order",
+        lead($"value", 1).over(window),
```

Comment:
how about something like `value + 1`, `value + order` that are not simple attributes?

Reply:
The current implement of dsl only accept column.